pkg/kube/inject/inject.go

```diff
@@ -847,3 +854,24 @@ func updateClusterEnvs(container *corev1.Container, newKVs map[string]string) {
 	}
 	container.Env = envVars
 }
+
+// GetProxyIDs returns the UID and GID to be used in the RunAsUser and RunAsGroup fields in the template
+// Inspects the namespace metadata for hints and fallbacks to the usual value of 1337.
+func GetProxyIDs(namespace *corev1.Namespace) (uid int64, gid int64) {
+	uid = constants.DefaultProxyUIDInt
+	gid = constants.DefaultProxyUIDInt
+
+	if namespace == nil {
+		return
+	}
+
+	// Check for OpenShift specifics and returns the max number in the range specified in the namespace annotation
```

Comment:
This explains what the code is doing but not _why_. Can you explain the openshift specific reasoning, in particular why "Max" is used?

Reply:
Sorry, there's no specifics here. We could pick the first value, the last, or any random value. At first I'd thought using random, but since it doesn't really matters, I avoided calling a random function and just picked the max.